src/EFCore/Metadata/Builders/EntityTypeBuilder.cs

```diff
@@ -412,17 +485,19 @@ public virtual IndexBuilder HasIndex([NotNull] params string[] propertyNames)
             Check.NotEmpty(navigationName, nameof(navigationName));
             Check.NotNull(buildAction, nameof(buildAction));
 
-            buildAction.Invoke(OwnsOneBuilder(new TypeIdentity(ownedType, (Model)Metadata.Model), navigationName));
+            buildAction(OwnsOneBuilder(new TypeIdentity(ownedType, (Model)Metadata.Model), sharedTypeClrType: null, navigationName));
             return this;
         }
 
-        private OwnedNavigationBuilder OwnsOneBuilder(in TypeIdentity ownedType, string navigationName)
+        private OwnedNavigationBuilder OwnsOneBuilder(in TypeIdentity ownedType, Type sharedTypeClrType, string navigationName)
         {
             InternalForeignKeyBuilder relationship;
             using (Builder.Metadata.Model.ConventionDispatcher.DelayConventions())
             {
                 relationship = ownedType.Type == null
-                    ? Builder.HasOwnership(ownedType.Name, navigationName, ConfigurationSource.Explicit)
+                    ? sharedTypeClrType == null
```

Comment:
```suggestion
                    ? targetClrType == null
```
It doesn't have to be a STET for the new methods other than `Entity`

Reply:
are you suggesting just variable name change (as the clr type does not necessarily have to be STET) or something bigger?

Reply:
Just name change for now